src/SFA.DAS.QnA.Api.UnitTests/ApplicationDataControllerTests/WhenPostingApplicationData.cs

```diff
@@ -0,0 +1,78 @@
+﻿using FluentAssertions;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+using NSubstitute;
+using NUnit.Framework;
+using SFA.DAS.QnA.Api.Controllers;
+using SFA.DAS.QnA.Api.Types;
+using SFA.DAS.QnA.Application.Commands.SetApplicationData;
+using System;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using JsonSerializer = System.Text.Json.JsonSerializer;
+
+namespace SFA.DAS.QnA.Api.UnitTests.ApplicationDataControllerTests
+{
+    public class WhenPostingApplicationData
+    {
+        [Test]
+        public async Task And_MediatorCallFails_ThenNotFoundResultReturned()
+        {
+            var mediator = Substitute.For<IMediator>();
+            var controller = new ApplicationDataController(mediator);
+            var applicationData = File.ReadAllText("ApplicationDataControllerTests/test.json");
+            mediator.Send(Arg.Any<SetApplicationDataRequest>()).Returns(new HandlerResponse<string>() { Success = false, Message = "ApplicationData does not validated against the Project's Schema." });
+
+            var result = await controller.Post(Guid.NewGuid(), applicationData);
+
+            result.Result.Should().BeOfType<NotFoundObjectResult>();
+        }
+
+        [Test]
+        public async Task And_MediatorCallIsSuccessful_ThenDeserializedResultReturned()
+        {
+            var mediator = Substitute.For<IMediator>();
+            var controller = new ApplicationDataController(mediator);
+            var applicationData = File.ReadAllText("ApplicationDataControllerTests/test.json");
+            var applicationDataResponse = JsonSerializer.Serialize(applicationData);
+            mediator.Send(Arg.Any<SetApplicationDataRequest>()).Returns(new HandlerResponse<string>(applicationDataResponse));
+
+            var actionResult = await controller.Post(Guid.NewGuid(), applicationData);
+            var resultAsJson = JsonDocument.Parse(actionResult.Value.ToString()).RootElement;
```

Comment:
The reason this test passes is because you're forcing it to pass by forcing the Post to return the fixed value assigned on line 39, thereby ignoring the actual content of the data sent to the Post. For example, this test would still pass if line 42 was changed to this:
```
var actionResult = await controller.Post(Guid.NewGuid(), @"{""animal"": ""panda""}");
```
We need to find a way of writing this test so that the above line would cause it to fail. Otherwise it's not really testing anything.

Reply:
Resolving because we've discussed an easier way of doing this